### PR TITLE
improve xbasic fixed string

### PR DESCRIPTION
**Pull request by serge-sans-paille**
Apply small string optimization on fixed string, and remove a useless template parameter.

### CODE REVIEW

File: include/xtl/xbasic_fixed_string.hpp [hunk 1]
```diff
@@ -67,7 +68,56 @@ namespace xtl
...
+              set_size(0);
+            }
+
+            fixed_small_string_storage_impl(T ptr[N], std::size_t size)
+                : m_buffer(ptr)
+            {
+              m_buffer[N - 1] = N - size;
+            }
+
+            auto& buffer()
```
Comment (JohanMabille) [inline include/xtl/xbasic_fixed_string.hpp:89]: Nitpicking: I think you can return T& instead of auto

File: include/xtl/xbasic_fixed_string.hpp [hunk 2]
```diff
@@ -67,7 +68,56 @@ namespace xtl
...
+            {
+                return m_buffer;
+            }
+
+            auto const & buffer() const
```
Comment (JohanMabille) [inline include/xtl/xbasic_fixed_string.hpp:94]: Same remark here

**Review by JohanMabille — CHANGES_REQUESTED** [review]
Thanks for this improvement!

General remark (formatting): we use 4 spaces indentation, and put opening curly braces on new lines.

File: include/xtl/xbasic_fixed_string.hpp (hunk 1 above)
Comment (serge-sans-paille) [reply include/xtl/xbasic_fixed_string.hpp:89]: I don't think so, `m_buffer` has type `T[N]`

**Comment by serge-sans-paille** [comment]
Formatting issue fixed.

File: include/xtl/xbasic_fixed_string.hpp (hunk 1 above)
Comment (JohanMabille) [reply include/xtl/xbasic_fixed_string.hpp:90]: My bad, I read too fast

**Review by JohanMabille — APPROVED** [review]

**Comment by JohanMabille** [comment]
Thanks!

**Pull request merged**